ferl_node:
    ros__parameters:
        setup:
            prefix: 'gen3'
            model_filename: "gen3"
            object_centers: 
                - "HUMAN_CENTER:[-0.2,-0.5,0.6]"
                - "LAPTOP_CENTER:[-0.8,0.0,0.0]"
                # HUMAN_CENTER: [-0.2,-0.5,0.6]
                # LAPTOP_CENTER: [0.0,-0.8,0.0]
            feat_list: ["laptop"]
            feat_weights: [0.0]
            # start: [104.2, 137.5, 183.8, 101.8, 224.2, 126.1, 310.8] # must adhere to joint limits
            # start: [4.5, 0.0, 0.0, 0.0, 1.5708, 0.0]
            # goal: [1.5, 0.0, 0.0, 0.0, 1.5708, 0.0]
            start: [0.698132, -2.0944, -1.0472, -3.14159, -0.837758, 0.0]
            goal: [-0.698132, -1.5973253, -1.6891297, -3.0077259, -2.23402, -0.008028515]
            # goal_pose: None #[-0.46513, 0.29041, 0.69497]
            T: 20.0
            timestep: 0.5
            save_dir: "/data/features/"
            # These constants have been tuned for the task tested.
            INTERACTION_TORQUE_THRESHOLD: [0.88414821, 17.22751856, -0.40134936,  6.23537946, -0.90013662, 1.32379884,  0.10218059]
            INTERACTION_TORQUE_EPSILON: [4.0, 5.0, 3.0, 4.0, 1.5, 1.5, 1.5]
            FEAT_RANGE: 
                - "table:0.98"
                - "coffee:1.0"
                - "laptop:0.3"
                - "human:0.3"
                - "efficiency:0.22"
                - "proxemics:0.3"
                - "betweenobjects:0.2"
                - "learned_feature:1.0"
            LF_dict:
                - "bet_data:5"
                - "sin:False"
                - "cos:False"
                - "rpy:False"
                - "lowdim:False"
                # - "EErot:True"
                - "EErot:False"
                - "noxyz:False"
                # - "norot:False"
                - "norot:True"
                - "noangles:True"
                - "6D_laptop:False"
                - "6D_human:False"
                - "9D_coffee:False"
                - "subspace_heuristic:False"
            CONFIDENCE_THRESHOLD: 1.0
            # N_QUERIES: 10
            N_QUERIES: 10
            nb_layers: 2
            nb_units: 64

        planner:
            # These settings have been tuned for trajopt planner.
            type: "trajopt"
            max_iter: 50
            num_waypts: 5

        controller:
            # These settings have been tuned for PID controller.
            type: "pid"
            p_gain: 5.0
            i_gain: 0.01
            d_gain: 0.1
            epsilon: 0.10
            max_cmd: 0.1

        learner:
            # These settings have been tuned for beta learner.
            type: "max"
            step_size: 1.0
            alpha: -0.01
            n: 5
            P_beta:
            - "table0:[1.83701582842, 0.0, 0.150583961407]"
            - "table1:[2.8, 0.0, 0.4212940611]"
            - "coffee0:[1.67451171875, 0.0, 0.05]"
            - "coffee1:[2.8169921875, 0.0, 0.3]"
            - "human0:[2.14693459432, 0.0, 0.227738059531]"
            - "human1:[5.0458984375, 0.0, 0.25]"


